junit-vintage-engine/src/main/java/org/junit/vintage/engine/discovery/MethodSelectorResolver.java

```diff
@@ -36,4 +36,28 @@ private void resolve(MethodSelector selector, TestClassCollector collector) {
 		collector.addFiltered(testClass, adapter(matchMethodDescription(methodDescription)));
 	}
 
+	private static Filter matchMethodDescription(final Description desiredDescription) {
```

Comment:
Please document where the code for `matchMethodDescription()` was copied from, including the rationale for why it was copied and modified.